--- origsrc/boost_1_40_0/boost/detail/interlocked.hpp	2008-07-12 04:55:08.000000000 -0500
+++ src/boost_1_40_0/boost/detail/interlocked.hpp	2009-08-27 23:41:34.067543900 -0500
@@ -92,7 +92,7 @@ extern "C" void* __cdecl _InterlockedExc
 # define BOOST_INTERLOCKED_EXCHANGE _InterlockedExchange
 # define BOOST_INTERLOCKED_EXCHANGE_ADD _InterlockedExchangeAdd
 
-#elif defined( WIN32 ) || defined( _WIN32 ) || defined( __WIN32__ ) || defined( __CYGWIN__ )
+#elif defined( WIN32 ) || defined( _WIN32 ) || defined( __WIN32__ )
 
 namespace boost
 {
